### ISSUE TITLE
Documentation is insufficient

### ISSUE THREAD

**Issue by ricardo-reis-1970** [gorakhargosh/watchdog#731]
Dear authors,

This library is setting a standard. We google anything regarding `python` and `file / folder monitoring` and we get swamped by references to your library. I have a big cauldron of spaghetti to refactor and `watchdog` will be there.

Or at least I _think_.

I'm trying to get a minimum PoC and it's proving very time consuming. The documentation is toothskin thin. Some examples I was able to mine from the repo don't even seem to be in the documentation.

This is a python library. I'm trying to use it as such. In my modest opinion, anything like a command-line tool _should_ take second chair to the front and centre mission of using the library in a project. This seems to not be the case, most unfortunately, but the opinion is indeed modest.

I'm trying to detect file changes, such as files created by medical devices and dumped in a shared folder, because that's how medical devices operate. Welcome to 1983! So, I'm trying to programmatically react to these events and already doing a lot, but I don't know, for instance, if I can pass to the Observer.schedule a list of folders, or whether the folders observed should all be descendants in the same tree.

In any case, I'm not asking for any particular explanation concerning my use case. This library is too good to go underdocumented like this. A reference is like a dictionary: you don't learn a foreign language by reading it.

How can I help?

**Comment by WolfgangFahl** [comment]
For a tool called watchdog using a polling loop as the standard example should not be the default. https://stackoverflow.com/questions/59843272/watchdog-as-background-thread-python explains a threading solution but that looks complicated and complicated things tend to be wrong ...

**Comment by pawamoy** [comment]
Hmmm more than that, it seems the documentation is currently broken? https://python-watchdog.readthedocs.io/en/v0.10.3/api.html#watchdog-events doesn't show any contents, while a previous version does: https://pythonhosted.org/watchdog/api.html#module-watchdog.events

**Comment by pawamoy** [comment]
Ah, it seems the docs moved to https://pythonhosted.org/watchdog/ indeed, but then they're for version 0.8.2, and I see *a lot* of more recent versions in the repository's tags 😕

**Comment by nicbou** [comment]
I agree. The documentation makes it very hard to use watchdog for more than the simplest use cases. For example, they don't explain what tricks are.

As a new user, I faced these issues, and could not find a solution after hours of Googling, reading issues, and reading the source code:

- Common tasks are not even mentioned:
    - Debouncing (preventing many rebuilds after multiple files change at once)
    - Preventing multiple rebuilds happening at the same time (files changing while a rebuild is in progress)
